cloud/scope/powervs_machine.go

```diff
@@ -1060,6 +1060,35 @@ func (m *PowerVSMachineScope) CreateVPCLoadBalancerPoolMember() (*vpcv1.LoadBala
 					}
 				}
 			}
+
+			for _, al := range lb.AdditionalListeners {
+				var shouldAdd bool
+
+				machineType, ok := m.IBMPowerVSMachine.ObjectMeta.Labels["powervs.cluster.x-k8s.io/node-type"]
+				m.V(3).Info("Testing additional listener", "Add", al.Add, "Port", al.Port, "machineType", machineType, "ok", ok)
+
+				if al.Add == "" || !ok {
+					continue
+				}
+
+				switch al.Add {
+				case infrav1beta2.VPCLoadBalancerPoolAddAll:
+					shouldAdd = true
+				case infrav1beta2.VPCLoadBalancerPoolAddBootstrapOnly:
+					shouldAdd = machineType == "bootstrap"
+				case infrav1beta2.VPCLoadBalancerPoolAddMastersOnly:
+					shouldAdd = machineType == "master"
+				default:
```

Comment:
Don't we need to set `shouldAdd = true` here also, when it's not set? Also can you please add UT covering this new logic?

Reply:
Yes, The default case workflow should be like the existing behavior, It should not break the existing flow.

Reply:
The default case throws an error so it does not need shouldAdd to be set.
What is UT?  Do you have an example of how to implement it?

Reply:
first lets focus on finalising the API spec, see my comments in the api/